Okta.Xamarin/Okta.Xamarin/IOktaStateManager.cs

```diff
@@ -16,38 +16,125 @@ namespace Okta.Xamarin
     /// </summary>
     public interface IOktaStateManager
     {
+        /// <summary>
+        /// Gets the access token.
+        /// </summary>
         string AccessToken { get; }
 
+        /// <summary>
+        /// Gets or sets the configuration.
+        /// </summary>
         IOktaConfig Config { get; set; }
 
+        /// <summary>
+        /// Gets or sets the client.
+        /// </summary>
+        IOidcClient Client { get; set; }
+
+        /// <summary>
+        /// Gets the expiration time.
+        /// </summary>
         DateTime? Expires { get; }
 
+        /// <summary>
+        /// Gets the id token.
+        /// </summary>
         string IdToken { get; }
 
+        /// <summary>
+        /// Gets a value indicating whether or not there is a current non-expired <see cref="AccessToken"/>, indicating the user is currently successfully authenticated.
+        /// </summary>
         bool IsAuthenticated { get; }
 
-        HttpResponseMessage LastApiResponse { get; set; }
+        /// <summary>
+        /// Gets the last response received from the API. Primarily for debugging.
+        /// </summary>
+        HttpResponseMessage LastApiResponse { get; }
 
+        /// <summary>
+        /// Gets the refresh token.
+        /// </summary>
         string RefreshToken { get; }
 
+        /// <summary>
+        /// Gets the scope.
+        /// </summary>
         string Scope { get; }
 
+        /// <summary>
+        /// Gets the type of the AccessToken, for example "Bearer".
+        /// </summary>
```

Comment:
What's the difference between TokenType vs TokenKind? I noticed we have 2 diff ways to categorize tokens.

Reply:
TokenType is what the server API returns describing the token for use as an `Authorization` header; the only value I've seen so far is "Bearer" but I assume other valid values would include "SSWS" or "basic".  To differentiate between the kind of token used in an `Authorization` header and the tokens that a developer will have related to the user's authentication state I created the TokenKind enum; valid values are `IdToken`, `AccessToken` and `RefreshToken`.  Because these are categorically different I wanted to disambiguate.